independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/event/fire/EventFireTest.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.arc.test.event.fire;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import io.quarkus.arc.test.event.select.BreakInEvent;
+import io.quarkus.arc.test.event.select.NotABindingType;
+import io.quarkus.arc.test.event.select.SecurityEvent;
+import io.quarkus.arc.test.event.select.SecurityEvent_Illegal;
+import io.quarkus.arc.test.event.select.SecuritySensor;
+import io.quarkus.arc.test.event.select.SystemTest;
+
+public class EventFireTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(BreakInEvent.class, NotABindingType.class,
+            SecurityEvent.class, SecurityEvent_Illegal.class,
+            SecuritySensor.class, SystemTest.class);
+
+    @Test
+    public <T> void testEventFireThrowsExceptionIfEventTypeHasTypeVariable() {
+        try {
```

Comment:
Let's also use the `org.junit.jupiter.api.Assertions.assertThrows(Class<T>, Executable)` instead to make it more readable...

Reply:
Fair enough, I've also changed other tests in `EventSelectTest` to use this instead.